traitlets/traitlets.py

```diff
@@ -1644,6 +1644,21 @@ def trait_events(cls, name=None):
                         events[k] = v
         return events
 
+    def clear(self, *names, **metadata):
+        """Remove the values of listed traits or those with certain metadata.
+
+        If no metadata or names are given, then all trait values are cleared.
+        """
+        if not metadata and not names:
+            names = self.trait_names(**metadata) + list(names)
```

Comment:
Thus looks like it belongs to the case where both have values not to the case where both have no values